### PR TITLE
GDKotlin Rework: Improve commandline, json and export.

### PR DESCRIPTION
**Pull request by CedNaru**
Fix some issues with the previous configuration/command-line PR.

Regarding the export, I fixed the Linux issue, replaced all the strings with the same constants used in GDKotlin.
I also changed the JVM type exported for desktop. It doesn't use the same one as the editor anymore.
By default it uses the JVM, but you can add a `export-graal-native-image` feature to make it export graal instead.

### CODE REVIEW

File: docs/src/doc/user-guide/advanced/commandline-args-and-configuration.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+The following arguments can be either used in the command line or the `godot_kotlin_configuration.json` file at the root
+of the project to customize the behaviour of the Godot Kotlin/JVM binding.
+
+Note that in case the same argument is used in both JSON and command-line, the command-line argument got the priority.
```
Comment (chippmann) [inline docs/src/doc/user-guide/advanced/commandline-args-and-configuration.md:4]: Would write this in an info block

File: harness/tests/godot_kotlin_configuration.json [hunk 2]
```diff
@@ -1 +1,14 @@
-{"vm_type":"jvm"}
+{
+  "debug_address": "*",
+  "debug_port": 5005,
+  "disable_closing_leaks_warning": false,
+  "disable_gc": false,
+  "force_gc": false,
+  "jmx_port": -1,
+  "jvm_args": "",
+  "max_string_size": -1,
```
Comment (chippmann) [inline harness/tests/godot_kotlin_configuration.json:9]: NIT; still -1? (I'm fine if you prefer it over 512 here)

File: src/lifecycle/jvm_user_configuration.h [hunk 3]
```diff
@@ -16,12 +18,13 @@ static constexpr const char* DISABLE_LEAK_WARNING_JSON_IDENTIFIER {"disable_clos
...
 // COMMAND LINE IDENTIFIER
-static constexpr const char* VM_TYPE_CMD_IDENTIFIER {"--java-vm-type"};
+static constexpr const char* VM_TYPE_CMD_IDENTIFIER {"--jvm-vm-type"};
+static constexpr const char* USE_DEBUG_CMD_IDENTIFIER {"--jvm-use_debug"};
 static constexpr const char* DEBUG_PORT_CMD_IDENTIFIER {"--jvm-debug-port"};
 static constexpr const char* DEBUG_ADDRESS_CMD_IDENTIFIER {"--jvm-debug-address"};
 static constexpr const char* WAIT_FOR_DEBUGGER_CMD_IDENTIFIER {"--wait-for-debugger"};
 static constexpr const char* JMX_PORT_CMD_IDENTIFIER {"--jvm-jmx-port"};
-static constexpr const char* MAX_STRING_SIZE_CMD_IDENTIFIER {"--jvm-to-engine-max-string-size"};
+static constexpr const char* MAX_STRING_SIZE_CMD_IDENTIFIER {"--jvm-max_string_size"};
```
Comment (chippmann) [inline src/lifecycle/jvm_user_configuration.h:27]: everywhere its - except here where its _?

File: src/kotlin_editor_export_plugin.cpp [hunk 4]
```diff
@@ -9,97 +9,91 @@
...
-        Error error;
-        Ref<DirAccess> dir_access {DirAccess::open(export_dir, &error)};
-        if (error == OK) {
             if (is_osx_export) {
                 bool is_arm64 {p_features.has("arm64")};
                 bool is_x64 {p_features.has("x86_64")};
 
                 if (!is_arm64 && !is_x64) {
-                    add_macos_plugin_file(vformat("res://%s", EMBEDDED_JRE_DIRECTORY));
+                    add_macos_plugin_file(String(RES_DIRECTORY).path_join(EMBEDDED_JRE_DIRECTORY));
```
Comment (chippmann) [inline src/kotlin_editor_export_plugin.cpp:42]: more of a question: will this not result in: `res:///path` (note the third /)?

File: harness/tests/godot_kotlin_configuration.json (hunk 2 above)
Comment (CedNaru) [reply harness/tests/godot_kotlin_configuration.json:9]: What do you mean still -1 ?
You disagreed with the default value being 0 as it people can want JNI strings only, so we decided to set it to -1 instead.

File: src/lifecycle/jvm_user_configuration.h (hunk 3 above)
Comment (CedNaru) [reply src/lifecycle/jvm_user_configuration.h:27]: Small mistake

File: src/kotlin_editor_export_plugin.cpp (hunk 4 above)
Comment (CedNaru) [reply src/kotlin_editor_export_plugin.cpp:42]: `path_join` is a function explicitly made to handle such case.
https://docs.godotengine.org/fr/4.x/classes/class_string.html#class-string-method-path-join

File: harness/tests/godot_kotlin_configuration.json (hunk 2 above)
Comment (chippmann) [reply harness/tests/godot_kotlin_configuration.json:9]: fair

**Review by chippmann — APPROVED** [review]

File: src/kotlin_editor_export_plugin.cpp [hunk 5]
```diff
@@ -9,97 +9,90 @@
...
+                }
+                if (dir_access->copy_dir(jre_dir, target_dir) != OK) {
+                    LOG_ERROR(vformat("Cannot copy %s folder to export folder, please make sure you created a JRE directory at the root of your project using jlink.", EMBEDDED_JRE_AMD_DIRECTORY)
+                    );
+                }
+            }
+
+        } else if (export_graal) {
+            if (is_windows_export) {
+                files_to_add.push_back(String(BUILD_DIRECTORY) + WIND0WS_GRAAL_NATIVE_IMAGE_FILE);
```
Comment (piiertho) [inline src/kotlin_editor_export_plugin.cpp:63]: typo => WIND0WS_GRAAL_NATIVE_IMAGE_FILE => WINDOWS_GRAAL_NATIVE_IMAGE_FILE

**Review by piiertho — APPROVED** [review]

**Pull request merged**